src/ui/features/premium/getPremiumStatus.ts

```diff
@@ -0,0 +1,59 @@
+import { normalizeAddress } from 'src/shared/normalizeAddress';
+import type { WalletMeta } from 'src/modules/zerion-api/requests/wallet-get-meta';
+import { useAllSignerOrHwAddresses } from 'src/ui/shared/requests/useAllExistingAddresses';
+import { useWalletsMetaByChunks } from 'src/ui/shared/requests/useWalletsMetaByChunks';
+import { useMemo } from 'react';
+
+function getPremiumStatus({
+  normalizedAddress,
+  walletsMeta,
+}: {
+  normalizedAddress?: string;
+  walletsMeta: WalletMeta[];
+}) {
+  return walletsMeta.some((meta) => {
+    if (
+      meta.membership.premium &&
+      meta.membership.premium?.plan !== 'Restricted'
+    ) {
+      return true;
+    }
+    if (
+      normalizedAddress &&
+      Boolean(meta.membership.premium) &&
+      normalizeAddress(meta.address) === normalizedAddress
+    ) {
+      return true;
```

Comment:
Not sure I get the logic
If any other address being premium is enough, why pass current address at all?
But also if address is current then "restricted" value doesn't matter?

If the described logic is correct then I think it requires a comment

Reply:
Done!
I hope this will help a bit :)